src/main/java/com/fasterxml/jackson/core/util/JsonParserSequence.java

```diff
@@ -77,7 +77,7 @@ public static JsonParserSequence createFlattened(boolean checkForExistingToken,
             return new JsonParserSequence(checkForExistingToken,
                     new JsonParser[] { first, second });
         }
-        ArrayList<JsonParser> p = new ArrayList<JsonParser>();
+        ArrayList<JsonParser> p = new ArrayList<JsonParser>(10);
```

Comment:
Default to size 10? :) Any particular reason to this number? :)

Reply:
It's the normal JDK default before the "start from size 0" change.

Reply:
Um ok, so unfortunately not anymore, JDK 8 and up does not default to 10

```
An ArrayList has an internal array to store the list elements.
There is a difference between the two constructor calls in Java 7 and 8:
If you do new ArrayList<>(0) the ArrayList creates a new Object array of size 0.
If you do new ArrayList<>() the ArrayList uses a static empty Object array of size 0 and switches to an own private array once you add items to the list.

The Javadoc of the default ArrayList constructor seems to contradict this.
```

Can we set this back to 0 please, 2.10 targets JDK 8

Reply:
There would be no point in setting it to 0 in this instance since there's an "add" command shortly after. So might as well construct it to begin with. The JDK does these optimizations for cases where it "might not" get used.

Reply:
Handing over to @cowtowncoder

I'm trying to avoid the initial pitfall, correct initializing with 0 will skip the static performance boost, 
Reason it was put in this way is because instantiating at 10 and popping in 40 items results in a huge performance decrease before it stabilizes out again.
I view the push to instantiate with 10 as a regression for JDK 8

![image](https://user-images.githubusercontent.com/5367513/51751320-43358700-20bd-11e9-9803-a98cbbbbbe74.png)

![image](https://user-images.githubusercontent.com/5367513/51751333-4af52b80-20bd-11e9-8795-2fe76ee6978a.png)

![image](https://user-images.githubusercontent.com/5367513/51751374-66603680-20bd-11e9-9e76-24ae13c8684c.png)

Reply:
I'm not sure what you're referring to, sorry. Starting at 0 just means it'll slowly grow to 10 anyhow. You can check the latest source code if needed. The only difference is 1 starts at 0 `int newCapacity = oldCapacity + (oldCapacity >> 1);` OR starts at 10 and grows the same. If you're going to grow to 40 as per your example - they'll both cross paths. The only optimization 0 can be better than 10 is if it's for sure < 10.